ROMIO: fix bug when a write request is larger than INT_MAX (4-byte int)

wkliao opened the pull request:
Assuming ADIO_WriteContig() can only write a maximum of INT_MAX bytes at a time. This small fix makes sure the max does not exceed. Below is a short test program with error message embedded to be run on one MPI process. Similar issue occurs at the read end. I will create a separate pull request for read.
```
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>

#define ERR if (err != MPI_SUCCESS) { \
        int errorStringLen; \
        char errorString[MPI_MAX_ERROR_STRING]; \
        MPI_Error_string(err, errorString, &errorStringLen); \
        printf("Error at line %d: %s\n",__LINE__, errorString); \
    }

#define TWO_G  2147483648
#define ONE_G  1073741824

int main(int argc, char **argv)
{
    int err, cmode=MPI_MODE_CREATE|MPI_MODE_RDWR;
    char *buf;
    int blocklen[3] = {10, ONE_G-1024, ONE_G+1024};
    MPI_Aint disp[3] = {0, 1024, ONE_G};
    MPI_Datatype dtype;
    MPI_Info info;
    MPI_File fh;
    MPI_Status status;

    MPI_Init(&argc, &argv);
    buf = (char*) calloc(TWO_G+1024,1);

    /* Disable data sieving hint causes MPI_File_write(): Assertion failed in file
     * adio/common/ad_write_str_naive.c at line 308: req_len == (int) req_len
     */
    MPI_Info_create(&info);
    MPI_Info_set(info, "romio_ds_write", "disable");

    err = MPI_File_open(MPI_COMM_WORLD, "testfile", cmode, info, &fh); ERR

    err = MPI_Type_create_hindexed(3, blocklen, disp, MPI_BYTE, &dtype); ERR
    err = MPI_Type_commit(&dtype); ERR

    err = MPI_File_set_view(fh, 0, MPI_BYTE, dtype, "native", MPI_INFO_NULL); ERR
    err = MPI_File_write(fh, buf, 1, dtype, &status); ERR

    err = MPI_File_close(&fh); ERR
    MPI_Type_free(&dtype);
    MPI_Info_free(&info);
    free(buf);

    MPI_Finalize();
    return 0;
}
```

raffenet commented:
@roblatham00 this looks correct to me, unless we want to change the type signature of Read/WriteContig functions. What do you think?

pavanbalaji commented:
@wkliao can this be rebased on the latest `mpich/master`?

pavanbalaji commented:
This is superseded by #2977.

wkliao commented:
Hi Pavan
I am struggling with an git issue that is unique to MPICH, thus could not rebase my changes.
I keep getting the following message when running "git status" even through I did not touch those two files.
```
	modified:   examples/examples.sln
	modified:   src/util/logging/rlog/rlogtools.sln
```
Google search points to the following git attribute used by MPICH.
```
# Declare files that will always have CRLF line endings on checkout.
*.sln text eol=crlf
```
This setting is for Windows environment, but I am testing MPICH on a Redhat.
Is there a way to avoid this problem?

pavanbalaji commented:
Hi @wkliao.  Yes, it's a mess to rebase because of some large changes that we made recently.  I've taken care of it for most of your patches.  I'm working on one more.  I'll finish that after lunch.
